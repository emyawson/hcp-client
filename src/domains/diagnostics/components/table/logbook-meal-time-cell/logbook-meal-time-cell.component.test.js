import { shallow } from 'enzyme';
import React from 'react';

import { LogbookMealTimeCell } from '.';

const mockProps = {
  hasBeforeAndAfterIntervals: true,
  mealTimeMatrix: [
    {
      after: {},
      before: {
        afterMeal: false,
        beforeMeal: false,
        date: 'Wed Dec 27 2017 17:48:00 GMT-0500 (EST)',
        glucose: 146,
        bolus: null,
        carbohydrates: null,
      },
    },
  ],
  numberOfRowsWithContent: 2,
  numberOfTotalRows: 4,
};

describe('LogbookMealTimeCell test suite', () => {
  test('it renders correctly', () => {
    const wrapper = shallow(<LogbookMealTimeCell {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
