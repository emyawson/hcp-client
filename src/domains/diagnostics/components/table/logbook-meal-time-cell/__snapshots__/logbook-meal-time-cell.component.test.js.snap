// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogbookMealTimeCell test suite it renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    hasBeforeAndAfterIntervals={true}
    mealTimeMatrix={
      Array [
        Object {
          "after": Object {},
          "before": Object {
            "afterMeal": false,
            "beforeMeal": false,
            "bolus": null,
            "carbohydrates": null,
            "date": "Wed Dec 27 2017 17:48:00 GMT-0500 (EST)",
            "glucose": 146,
          },
        },
      ]
    }
    numberOfRowsWithContent={2}
    numberOfTotalRows={4}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "function",
    "props": Object {
      "borderRight": true,
      "children": Array [
        Array [
          <LogbookCellBlocksContainer
            borderBottom={true}
          >
            <Unknown
              alternateFill={false}
              borderRight={false}
              flex={1}
            >
              <Unknown
                aboveTargetRange={undefined}
                belowTargetRange={undefined}
                hypoSymptoms={undefined}
                unit="glucose"
                value={146}
              >
                146
              </Unknown>
            </Unknown>
            <Unknown
              alternateFill={false}
              borderRight={false}
              flex={1}
            >
              <Unknown
                aboveTargetRange={undefined}
                belowTargetRange={undefined}
                hypoSymptoms={undefined}
                unit="carbohydrates"
                value={null}
              >
                <span>
                   
                </span>
              </Unknown>
            </Unknown>
            <Unknown
              alternateFill={false}
              borderRight={true}
              flex={1}
            >
              <Unknown
                aboveTargetRange={undefined}
                belowTargetRange={undefined}
                hypoSymptoms={undefined}
                unit="bolus"
                value={null}
              >
                <span>
                   
                </span>
              </Unknown>
            </Unknown>
            <Unknown
              alternateFill={true}
              borderRight={false}
              flex={1}
            >
              <Unknown
                aboveTargetRange={undefined}
                belowTargetRange={undefined}
                hypoSymptoms={undefined}
                unit="glucose"
                value={undefined}
              >
                <span>
                   
                </span>
              </Unknown>
            </Unknown>
            <Unknown
              alternateFill={true}
              borderRight={false}
              flex={1}
            >
              <Unknown
                aboveTargetRange={undefined}
                belowTargetRange={undefined}
                hypoSymptoms={undefined}
                unit="bolus"
                value={undefined}
              >
                <span>
                   
                </span>
              </Unknown>
            </Unknown>
          </LogbookCellBlocksContainer>,
        ],
        <Unknown
          alternateFill={undefined}
          day={undefined}
          numberOfCells={5}
          numberOfRows={2}
          showBorder={true}
        />,
      ],
      "mouseClickHandler": undefined,
      "pointerCursor": true,
      "width": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "day row mealtime row - undefined - 0",
        "nodeType": "class",
        "props": Object {
          "borderBottom": true,
          "children": Array [
            <Unknown
              alternateFill={false}
              borderRight={false}
              flex={1}
            >
              <Unknown
                aboveTargetRange={undefined}
                belowTargetRange={undefined}
                hypoSymptoms={undefined}
                unit="glucose"
                value={146}
              >
                146
              </Unknown>
            </Unknown>,
            <Unknown
              alternateFill={false}
              borderRight={false}
              flex={1}
            >
              <Unknown
                aboveTargetRange={undefined}
                belowTargetRange={undefined}
                hypoSymptoms={undefined}
                unit="carbohydrates"
                value={null}
              >
                <span>
                   
                </span>
              </Unknown>
            </Unknown>,
            <Unknown
              alternateFill={false}
              borderRight={true}
              flex={1}
            >
              <Unknown
                aboveTargetRange={undefined}
                belowTargetRange={undefined}
                hypoSymptoms={undefined}
                unit="bolus"
                value={null}
              >
                <span>
                   
                </span>
              </Unknown>
            </Unknown>,
            <Unknown
              alternateFill={true}
              borderRight={false}
              flex={1}
            >
              <Unknown
                aboveTargetRange={undefined}
                belowTargetRange={undefined}
                hypoSymptoms={undefined}
                unit="glucose"
                value={undefined}
              >
                <span>
                   
                </span>
              </Unknown>
            </Unknown>,
            <Unknown
              alternateFill={true}
              borderRight={false}
              flex={1}
            >
              <Unknown
                aboveTargetRange={undefined}
                belowTargetRange={undefined}
                hypoSymptoms={undefined}
                unit="bolus"
                value={undefined}
              >
                <span>
                   
                </span>
              </Unknown>
            </Unknown>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "day row mealtime cell - undefined - unit - 0",
            "nodeType": "function",
            "props": Object {
              "alternateFill": false,
              "borderRight": false,
              "children": <Unknown
                aboveTargetRange={undefined}
                belowTargetRange={undefined}
                hypoSymptoms={undefined}
                unit="glucose"
                value={146}
              >
                146
              </Unknown>,
              "flex": 1,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "function",
              "props": Object {
                "aboveTargetRange": undefined,
                "belowTargetRange": undefined,
                "children": 146,
                "hypoSymptoms": undefined,
                "unit": "glucose",
                "value": 146,
              },
              "ref": null,
              "rendered": 146,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "day row mealtime cell - undefined - unit - 1",
            "nodeType": "function",
            "props": Object {
              "alternateFill": false,
              "borderRight": false,
              "children": <Unknown
                aboveTargetRange={undefined}
                belowTargetRange={undefined}
                hypoSymptoms={undefined}
                unit="carbohydrates"
                value={null}
              >
                <span>
                   
                </span>
              </Unknown>,
              "flex": 1,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "function",
              "props": Object {
                "aboveTargetRange": undefined,
                "belowTargetRange": undefined,
                "children": <span>
                   
                </span>,
                "hypoSymptoms": undefined,
                "unit": "carbohydrates",
                "value": null,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": " ",
                },
                "ref": null,
                "rendered": " ",
                "type": "span",
              },
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "day row mealtime cell - undefined - unit - 2",
            "nodeType": "function",
            "props": Object {
              "alternateFill": false,
              "borderRight": true,
              "children": <Unknown
                aboveTargetRange={undefined}
                belowTargetRange={undefined}
                hypoSymptoms={undefined}
                unit="bolus"
                value={null}
              >
                <span>
                   
                </span>
              </Unknown>,
              "flex": 1,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "function",
              "props": Object {
                "aboveTargetRange": undefined,
                "belowTargetRange": undefined,
                "children": <span>
                   
                </span>,
                "hypoSymptoms": undefined,
                "unit": "bolus",
                "value": null,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": " ",
                },
                "ref": null,
                "rendered": " ",
                "type": "span",
              },
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "day row mealtime cell - undefined - unit - 3",
            "nodeType": "function",
            "props": Object {
              "alternateFill": true,
              "borderRight": false,
              "children": <Unknown
                aboveTargetRange={undefined}
                belowTargetRange={undefined}
                hypoSymptoms={undefined}
                unit="glucose"
                value={undefined}
              >
                <span>
                   
                </span>
              </Unknown>,
              "flex": 1,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "function",
              "props": Object {
                "aboveTargetRange": undefined,
                "belowTargetRange": undefined,
                "children": <span>
                   
                </span>,
                "hypoSymptoms": undefined,
                "unit": "glucose",
                "value": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": " ",
                },
                "ref": null,
                "rendered": " ",
                "type": "span",
              },
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "day row mealtime cell - undefined - unit - 4",
            "nodeType": "function",
            "props": Object {
              "alternateFill": true,
              "borderRight": false,
              "children": <Unknown
                aboveTargetRange={undefined}
                belowTargetRange={undefined}
                hypoSymptoms={undefined}
                unit="bolus"
                value={undefined}
              >
                <span>
                   
                </span>
              </Unknown>,
              "flex": 1,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "function",
              "props": Object {
                "aboveTargetRange": undefined,
                "belowTargetRange": undefined,
                "children": <span>
                   
                </span>,
                "hypoSymptoms": undefined,
                "unit": "bolus",
                "value": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": " ",
                },
                "ref": null,
                "rendered": " ",
                "type": "span",
              },
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "alternateFill": undefined,
          "day": undefined,
          "numberOfCells": 5,
          "numberOfRows": 2,
          "showBorder": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "function",
      "props": Object {
        "borderRight": true,
        "children": Array [
          Array [
            <LogbookCellBlocksContainer
              borderBottom={true}
            >
              <Unknown
                alternateFill={false}
                borderRight={false}
                flex={1}
              >
                <Unknown
                  aboveTargetRange={undefined}
                  belowTargetRange={undefined}
                  hypoSymptoms={undefined}
                  unit="glucose"
                  value={146}
                >
                  146
                </Unknown>
              </Unknown>
              <Unknown
                alternateFill={false}
                borderRight={false}
                flex={1}
              >
                <Unknown
                  aboveTargetRange={undefined}
                  belowTargetRange={undefined}
                  hypoSymptoms={undefined}
                  unit="carbohydrates"
                  value={null}
                >
                  <span>
                     
                  </span>
                </Unknown>
              </Unknown>
              <Unknown
                alternateFill={false}
                borderRight={true}
                flex={1}
              >
                <Unknown
                  aboveTargetRange={undefined}
                  belowTargetRange={undefined}
                  hypoSymptoms={undefined}
                  unit="bolus"
                  value={null}
                >
                  <span>
                     
                  </span>
                </Unknown>
              </Unknown>
              <Unknown
                alternateFill={true}
                borderRight={false}
                flex={1}
              >
                <Unknown
                  aboveTargetRange={undefined}
                  belowTargetRange={undefined}
                  hypoSymptoms={undefined}
                  unit="glucose"
                  value={undefined}
                >
                  <span>
                     
                  </span>
                </Unknown>
              </Unknown>
              <Unknown
                alternateFill={true}
                borderRight={false}
                flex={1}
              >
                <Unknown
                  aboveTargetRange={undefined}
                  belowTargetRange={undefined}
                  hypoSymptoms={undefined}
                  unit="bolus"
                  value={undefined}
                >
                  <span>
                     
                  </span>
                </Unknown>
              </Unknown>
            </LogbookCellBlocksContainer>,
          ],
          <Unknown
            alternateFill={undefined}
            day={undefined}
            numberOfCells={5}
            numberOfRows={2}
            showBorder={true}
          />,
        ],
        "mouseClickHandler": undefined,
        "pointerCursor": true,
        "width": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "day row mealtime row - undefined - 0",
          "nodeType": "class",
          "props": Object {
            "borderBottom": true,
            "children": Array [
              <Unknown
                alternateFill={false}
                borderRight={false}
                flex={1}
              >
                <Unknown
                  aboveTargetRange={undefined}
                  belowTargetRange={undefined}
                  hypoSymptoms={undefined}
                  unit="glucose"
                  value={146}
                >
                  146
                </Unknown>
              </Unknown>,
              <Unknown
                alternateFill={false}
                borderRight={false}
                flex={1}
              >
                <Unknown
                  aboveTargetRange={undefined}
                  belowTargetRange={undefined}
                  hypoSymptoms={undefined}
                  unit="carbohydrates"
                  value={null}
                >
                  <span>
                     
                  </span>
                </Unknown>
              </Unknown>,
              <Unknown
                alternateFill={false}
                borderRight={true}
                flex={1}
              >
                <Unknown
                  aboveTargetRange={undefined}
                  belowTargetRange={undefined}
                  hypoSymptoms={undefined}
                  unit="bolus"
                  value={null}
                >
                  <span>
                     
                  </span>
                </Unknown>
              </Unknown>,
              <Unknown
                alternateFill={true}
                borderRight={false}
                flex={1}
              >
                <Unknown
                  aboveTargetRange={undefined}
                  belowTargetRange={undefined}
                  hypoSymptoms={undefined}
                  unit="glucose"
                  value={undefined}
                >
                  <span>
                     
                  </span>
                </Unknown>
              </Unknown>,
              <Unknown
                alternateFill={true}
                borderRight={false}
                flex={1}
              >
                <Unknown
                  aboveTargetRange={undefined}
                  belowTargetRange={undefined}
                  hypoSymptoms={undefined}
                  unit="bolus"
                  value={undefined}
                >
                  <span>
                     
                  </span>
                </Unknown>
              </Unknown>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "day row mealtime cell - undefined - unit - 0",
              "nodeType": "function",
              "props": Object {
                "alternateFill": false,
                "borderRight": false,
                "children": <Unknown
                  aboveTargetRange={undefined}
                  belowTargetRange={undefined}
                  hypoSymptoms={undefined}
                  unit="glucose"
                  value={146}
                >
                  146
                </Unknown>,
                "flex": 1,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "function",
                "props": Object {
                  "aboveTargetRange": undefined,
                  "belowTargetRange": undefined,
                  "children": 146,
                  "hypoSymptoms": undefined,
                  "unit": "glucose",
                  "value": 146,
                },
                "ref": null,
                "rendered": 146,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "day row mealtime cell - undefined - unit - 1",
              "nodeType": "function",
              "props": Object {
                "alternateFill": false,
                "borderRight": false,
                "children": <Unknown
                  aboveTargetRange={undefined}
                  belowTargetRange={undefined}
                  hypoSymptoms={undefined}
                  unit="carbohydrates"
                  value={null}
                >
                  <span>
                     
                  </span>
                </Unknown>,
                "flex": 1,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "function",
                "props": Object {
                  "aboveTargetRange": undefined,
                  "belowTargetRange": undefined,
                  "children": <span>
                     
                  </span>,
                  "hypoSymptoms": undefined,
                  "unit": "carbohydrates",
                  "value": null,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": " ",
                  },
                  "ref": null,
                  "rendered": " ",
                  "type": "span",
                },
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "day row mealtime cell - undefined - unit - 2",
              "nodeType": "function",
              "props": Object {
                "alternateFill": false,
                "borderRight": true,
                "children": <Unknown
                  aboveTargetRange={undefined}
                  belowTargetRange={undefined}
                  hypoSymptoms={undefined}
                  unit="bolus"
                  value={null}
                >
                  <span>
                     
                  </span>
                </Unknown>,
                "flex": 1,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "function",
                "props": Object {
                  "aboveTargetRange": undefined,
                  "belowTargetRange": undefined,
                  "children": <span>
                     
                  </span>,
                  "hypoSymptoms": undefined,
                  "unit": "bolus",
                  "value": null,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": " ",
                  },
                  "ref": null,
                  "rendered": " ",
                  "type": "span",
                },
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "day row mealtime cell - undefined - unit - 3",
              "nodeType": "function",
              "props": Object {
                "alternateFill": true,
                "borderRight": false,
                "children": <Unknown
                  aboveTargetRange={undefined}
                  belowTargetRange={undefined}
                  hypoSymptoms={undefined}
                  unit="glucose"
                  value={undefined}
                >
                  <span>
                     
                  </span>
                </Unknown>,
                "flex": 1,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "function",
                "props": Object {
                  "aboveTargetRange": undefined,
                  "belowTargetRange": undefined,
                  "children": <span>
                     
                  </span>,
                  "hypoSymptoms": undefined,
                  "unit": "glucose",
                  "value": undefined,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": " ",
                  },
                  "ref": null,
                  "rendered": " ",
                  "type": "span",
                },
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "day row mealtime cell - undefined - unit - 4",
              "nodeType": "function",
              "props": Object {
                "alternateFill": true,
                "borderRight": false,
                "children": <Unknown
                  aboveTargetRange={undefined}
                  belowTargetRange={undefined}
                  hypoSymptoms={undefined}
                  unit="bolus"
                  value={undefined}
                >
                  <span>
                     
                  </span>
                </Unknown>,
                "flex": 1,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "function",
                "props": Object {
                  "aboveTargetRange": undefined,
                  "belowTargetRange": undefined,
                  "children": <span>
                     
                  </span>,
                  "hypoSymptoms": undefined,
                  "unit": "bolus",
                  "value": undefined,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": " ",
                  },
                  "ref": null,
                  "rendered": " ",
                  "type": "span",
                },
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "alternateFill": undefined,
            "day": undefined,
            "numberOfCells": 5,
            "numberOfRows": 2,
            "showBorder": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
